sailor/utils/utils.py

```diff
@@ -16,3 +18,21 @@ def _is_non_string_iterable(obj):
     if issubclass(obj.__class__, str):
         return False
     return isinstance(obj, Iterable)
+
+
+class WarningAdapter(logging.LoggerAdapter):
+    """Allow a logger to convert warnings logs into real warnings to simplify logging setup for users."""
+
+    def __init__(self, logger, extra=None):
+        super().__init__(logger, extra or {})
+
+    def log_with_warning(self, msg, warning_stacklevel=1, warning_category=None):
+        """Delegate a warning call to the underlying logger and trigger a real warning with the same message."""
+        warnings.warn(msg, category=warning_category, stacklevel=warning_stacklevel + 2)
+        self.log(logging.WARNING, msg)
+
+    def log(self, level, msg, *args, **kwargs):
+        """Adjust the stacklevel parameter, because the logging adapter introduces additional stack layers."""
```

Comment:
If you only read this docstring you would think that the log function does not log something.

It should read

"Delegate a log call to LoggerAdapter.log, after adjusting the stacklevel for introduced stack layers by the adapter."

Reply:
Makes sense. :) I changed the docstring according to your suggestion (but leaving the last three words out to make the docstring fit to one line as needed for pydocstyle: "Delegate a log call to LoggerAdapter.log, after adjusting the stacklevel for introduced stack layers.")